### PR TITLE
Improve docker-compose configuration according to best practices:

### PR DESCRIPTION
**Pull request by wajda**
- Rename `docker-compose.yml` to `compose.yaml`
- Remove `version` top-level element
- Add `name` top-level element
- Replace `links` with `depends_on`
- Rename some services for clarity
- Services (except for UI) to communicate via hostnames and on the internal network, not to be routed via the host.

Thanks @korbelm for hints

### CODE REVIEW

File: docker/compose.yaml [hunk 1]
```diff
@@ -1,63 +1,71 @@
...
         catalina.sh run
       "
     environment:
-      SPLINE_DATABASE_CONNECTION_URL: 'arangodb://172.17.0.1:${ARANGO_DB_PORT}/spline'
+      SPLINE_DATABASE_CONNECTION_URL: 'arangodb://arangodb/spline'
       # by default /dev/random is used which may block
       CATALINA_OPTS: "-Dsecurerandom.source=file:/dev/./urandom -Djava.security.egd=file:/dev/./urandom"
-    links:
+    depends_on:
       - arangodb
```
Comment (cerveada) [inline docker/compose.yaml:43]: better solution could be:
```
  depends_on:
    db-init:
      condition: service_completed_successfully
```
logicaly `rest-server` depends on `db-init` which depends on `arangodb`

see similar solution here:
https://github.com/AbsaOSS/spline-openlineage/blob/master/docker/ol-integration-and-spline/docker-compose.yml

Comment (wajda) [reply docker/compose.yaml:43]: Makes sense. Will update. Thank you.

**Review by cerveada — APPROVED** [review]

**Comment by JinyLeeChina** [comment]
I tested composite and it reported an error
`service 'name' must be a mapping not a string`
Figure out this, but another will appear

**Comment by wajda** [comment]
@JinyLeeChina, I double checked and see no errors on my side. Please try to update your `docker` and `composite` plugin. If you have any error, please open a new issue and post the full log there. Thanks.

**Comment by cerveada** [comment]
It works for me as well

**Pull request merged**